src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -603,12 +602,12 @@ def ndim(self) -> int:
         return 1
 
     @property
-    @index_not_implemented()
-    def size(self) -> None:
+    @is_lazy_check
```

Comment:
Why have `is_lazy_check` only on this API?

Reply:
we don't need this right. We should do the check in `__len__`.

Reply:
sorry, what is this is_lazy_check annotation for ? the api here (shape and size) doesn't sounds like some API can be lazily done, right?

Reply:
the is_lazy_check is a check for whether or not we want to use a query for this API or if we want to use the locally cached native pandas object